src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java

```diff
@@ -188,10 +188,8 @@ protected void markSectionAsSpecial(Map<String, Boolean> isSectionSpecialMapping
         // indicate that section group covers the section
         // and mark that this section is special
         String sectionGroupParamName = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionGroupIndex;
-        if (specialSectionsInSectionGroups.get(sectionGroupParamName) == null) {
-            specialSectionsInSectionGroups.put(sectionGroupParamName, new ArrayList<String>());
-        }
-        specialSectionsInSectionGroups.get(sectionGroupParamName).add(sectionToMark);
+        specialSectionsInSectionGroups.computeIfAbsent(sectionGroupParamName, (String key) -> new ArrayList<String>())
```

Comment:
- `key` does not need to be typed; there is type inference here.
- `ArrayList<>()` (diamond constructor)